src/lib/handle-result.ts

```diff
@@ -12,12 +14,19 @@ import type {
 import type { BranchEntry, FunctionEntry, LineEntry, SectionSummary, Summary } from '../typings/file.js'
 
 export type FunctionMap = Map<string, FunctionEntry>
+export type FunctionLeaders = Map<number, FunctionLeaderEntry>
 
-export function handleResult(entry: EntryVariants, functionMap: FunctionMap, section: SectionSummary): boolean {
+export function handleResult(
+    entry: EntryVariants,
+    functionLeaders: FunctionLeaders,
+    functionMap: FunctionMap,
+    section: SectionSummary
+): boolean {
     switch (entry.variant) {
         case Variant.TestName:
             // creating a new test module, to prevent name conflicts simply clear the current FunctionMap
```

Comment:
It would be good to change this comment as well, either to mention the new `Map` as well, or (preferably) to remove the explicit mention and just describe why these maps need to be cleared.